pyccel/codegen/printing/ccode.py

```diff
@@ -83,6 +85,30 @@
     'NumpyArctanh': 'atanh',
 }
 
+numpy_ufunc_to_c_complex = {
+    'NumpyAbs'  : 'cabs',
+    'NumpyMin'  : 'minval',
+    'NumpyMax'  : 'maxval',
+    'NumpyFloor': 'floor',  # TODO: might require special treatment with casting
```

Comment:
Numpy's floor function throws an error if you try to call it on a complex number
```
In [6]: np.floor(2.4+3j)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-4e68486ab418> in <module>()
----> 1 np.floor(2.4+3j)

TypeError: ufunc 'floor' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

Reply:
Our code should therefore also throw a (readable) error. For an example of how to do this see:
https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/ccode.py#L214

Reply:
i was looking for this thank you